presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -772,7 +772,7 @@ public PlanNode visitLateralJoin(LateralJoinNode node, RewriteContext<Set<Symbol
             PlanNode subquery = context.rewrite(node.getSubquery(), context.get());
 
             // remove unused lateral nodes
-            if (subquery.getOutputSymbols().isEmpty() && node.getType() == LateralJoinNode.Type.INNER) {
+            if (subquery.getOutputSymbols().isEmpty() && node.getType() == LateralJoinNode.Type.INNER && ScalarQueryUtil.isScalar(subquery)) {
```

Comment:
You can remove lateral if:
1. `intersect(subquery.getOutputSymbols(), context.get).isEmtpy()`
and:
a. lateral is inner and right side is scalar (currently incorrectly implemented for filter)
b. lateral is left

add tests